packages/react-theme-provider/src/components/ThemeProvider/useThemeProvider.ts

```diff
@@ -37,18 +38,10 @@ export const useThemeProvider = (
   const localTheme = state.theme;
 
   state.theme = mergeThemes(parentTheme, localTheme);
-  state.style = React.useMemo(() => {
-    // TODO: should we consider insertion to head?
-    //       - how to modify, remove styles?
-    //       - SSR rendering
-
-    // TODO: what variables should be rendered? Merged or only changed?
-    // TODO: how we will proceed with Portals?
-    return {
-      ...state.style,
-      ...themeToCSSVariables(state.theme),
-    };
-  }, [state.style, state.theme]);
+  const themeClassName = useThemeStyleTag(state.theme, state.targetDocument);
 
+  state.themeClassName = themeClassName;
+  state.className = state.className || '';
+  state.className += ` ${themeClassName}`;
```

Comment:
nit: I would rather use `.join()` with `.filter()` to avoid an additional space:

```js
[undefined, "a"].filter(Boolean).join(" ")
// "a"
["", "a"].filter(Boolean).join(" ")
// "a"
["a", "a"].filter(Boolean).join(" ")
// "a a"
```